docs/source/internals/rest_api.rst

```diff
@@ -1,11 +1,272 @@
+
+.. code-block:: python
+
+    import aiida.restapi.common as common
+    from aiida import load_profile
+
+    CONFIG_DIR = common.__path__[0]
+
+    import click
+    @click.command()
+    @click.option('-P', '--port', type=click.INT, default=5000,
+        help='Port number')
+    @click.option('-H', '--hostname', default='127.0.0.1',
+        help='Hostname')
+    @click.option('-c','--config-dir','config',type=click.Path(exists=True), default=CONFIG_DIR,
+        help='the path of the configuration directory')
+    @click.option('--debug', 'debug', is_flag=True, default=False,
+        help='run app in debug mode')
+    @click.option('--wsgi-profile', 'wsgi_profile', is_flag=True, default=False,
+        help='to use WSGI profiler middleware for finding bottlenecks in web application')
+
+    def newendpoint(**kwargs):
+        """
+        runs the REST api
+        """
+        # Invoke the runner
+        run_api(App, NewApi, **kwargs)
+
+    # main program
+    if __name__ == '__main__':
+        """
+        Run the app with the provided options. For example:
+        python api.py --host=127.0.0.2 --port=6000
+        """
+
+        load_profile()
+        newendpoint()
+
+The `click package <https://click.palletsprojects.com/en/7.x/>`_ is used to provide a a nice command line interface to process the options and handle the default values to pass to the ``newendpoint`` function.
+
+The method ``run_api()`` accomplishes several functions:
+it couples the API to an instance of ``flask.Flask``, namely, the Flask fundamental class representing a web app. Consequently, the app is configured and, if required, hooked up.
+The spirit of ``run_api`` is to take all the ingredients to setup an API and use them to build up a command-line utility that serves to hook it up.
+
+It takes as inputs:
+
+    - the classes representing the Api and the App. We strongly suggest to pass to ``run_api()`` the :py:class:`aiida.restapi.api.App` class, inheriting from ``flask.Flask``, as it handles correctly AiiDA RESTApi-specific exceptions.
+
+    - positional arguments representing the command-line arguments/options, passed by the click function.
+      Types, defaults and help strings can be set in the ``@click.option`` definitions, and will be handled by the command line call.
+
+
+You should know few more things before using the script:
+
+    - If you want to customize further the error handling, you can take inspiration by looking at the definition of ``App`` and create your derived class ``NewApp(App)``.
+
+
+    - the supported command line options are identical to those of ``verdi restapi``.
+      Use ``verdi restapi --help`` for their full documentation.
+      If you want to add more options or modify the existing ones, create you custom runner taking inspiration from ``run_api``.
+
+It is time to run ``api.py``. Type in a terminal
+
+.. code-block:: bash
+
+    chmod +x api.py
+    ./api.py --host=127.0.0.2 --port=6000
```

Comment:
```suggestion
    ./api.py --hostname=127.0.0.2 --port=6000
```

Reply:
Ok I see which fix you mean - indeed I ran it without the CLI option.
It may be that 127.0.0.2 actually does not work - I'll check